Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneEventCollector/TrendMicroVisionOneEventCollector.py

```diff
@@ -0,0 +1,1151 @@
+        """
+        request_headers = headers or {
+            "Authorization": f"Bearer {self.api_key}"
+        }
+
+        url = next_link or f"{self.base_url}/{self.API_VERSION}{url_suffix}"
+        demisto.info(f'Sending the http request to {url=} with {params=}')
+
+        return self._http_request(
+            method=method,
+            full_url=url,
+            params=params,
+            headers=request_headers,
+        )
+
+    def get_logs(
+        self,
+        url_suffix: str,
+        method: str = 'GET',
+        params: dict | None = None,
+        headers: dict | None = None,
+        limit: int = DEFAULT_MAX_LIMIT,
+        next_link: str | None = None
+    ) -> tuple[List[dict], str | None]:
+        """
+        Implements a generic method with pagination to retrieve logs from trend micro vision one.
+
+        Args:
+            url_suffix (str): the URL suffix for the api endpoint.
+            method (str): the method of the api endpoint.
+            params (dict): query parameters for the api request.
+            headers (dict): any custom headers for the api request.
+            limit (str): the maximum number of events to retrieve.
+            next_link (str): the next link to continue pagination
+
+        Returns:
+            List[Dict]: a list of the requested logs.
+        """
+        logs: List[dict] = []
+
+        if next_link:
+            response = self.http_request(method=method, headers=headers, next_link=next_link)
+        else:
+            response = self.http_request(url_suffix=url_suffix, method=method, params=params, headers=headers)
+
+        current_items = response.get('items') or []
+        demisto.info(f'Received {current_items=} with {url_suffix=} and {params=}')
+        logs.extend(current_items)
+
+        while (new_next_link := response.get('nextLink')) and len(logs) < limit:
+            response = self.http_request(method=method, headers=headers, next_link=new_next_link)
+            current_items = response.get('items') or []
+            demisto.info(f'Received {current_items=} with {new_next_link=}')
+            logs.extend(current_items)
+
+        log_type, created_time_field = URL_SUFFIX_TO_EVENT_TYPE_AND_CREATED_TIME_FIELD[url_suffix]
+
+        if log_type not in (LogTypes.OBSERVED_ATTACK_TECHNIQUES.value, LogTypes.SEARCH_DETECTIONS.value) or not new_next_link:
+            # only limit cases where the logs are in ascending order in order not to lose part of events of nextLink
+            # if there is no next link, limit the logs as there is no chance we would lose events
+            logs = logs[:limit]
+
+        if url_suffix == UrlSuffixes.SEARCH_DETECTIONS.value:
+            for log in logs:
+                if log_time := log.get(created_time_field):
+                    log[created_time_field] = timestamp_to_datestring(
+                        timestamp=log_time, date_format=DATE_FORMAT, is_utc=True
```

Comment:
what are we changing here? Are we switching the time to utc where it might not have been before?

Reply:
we are switching timestamps (integers) to datestring